engineapi/proxy.go

```diff
@@ -109,6 +114,8 @@ func (p *Proxy) Close() {
 	if err := p.grpcClient.Close(); err != nil {
 		p.logger.Warn("failed to close engine client proxy")
 	}
+
+	p.ds.ProxyConnectionDecrease()
```

Comment:
Any potential issues if you decrease the ProxyConnection value regardless of the `Close` error?

Reply:
I think the only returned error is `"grpc: the client connection is closing"` which means we should also decrease that count. WDYT?

Ref https://github.com/longhorn/longhorn-manager/blob/master/vendor/google.golang.org/grpc/clientconn.go#L1040

Reply:
OK, looks fine here. Thank you.

Reply:
Good explanation. Can you make it a comment here?

Reply:
Updated.